drivers/timer/xtensa_sys_timer.c

```diff
@@ -42,11 +43,18 @@ static uint32_t ccount(void)
 static void ccompare_isr(const void *arg)
 {
 	ARG_UNUSED(arg);
+	uint32_t diff;
 
 	k_spinlock_key_t key = k_spin_lock(&lock);
 	uint32_t curr = ccount();
 	uint32_t dticks = (curr - last_count) / CYC_PER_TICK;
 
+	if (curr < last_count)
+		diff = (MAX_CYC - last_count) + curr;
+	else
+		diff = curr - last_count;
```

Comment:
This math looks off by a cycle.  MAX_CYC is 0xffffffff, not 0x100000000LL.  But the correct calculation is degenerate anyway, just write `curr - last_count`.  If both are uint32_t, the difference is guaranteed correct regardless of whether it wraps around.